src/main/kotlin/org/dif/diddoc/DIDDocResolver.kt

```diff
@@ -0,0 +1,14 @@
+package org.dif.diddoc
+
+/**
+ * Represents DID Doc resolver (https://www.w3.org/TR/did-core/#did-resolution).
+ */
+interface DIDDocResolver {
+    /**
+     * Resolves a DID document by the given DID.
+     * @param did a DID to be resolved.
+     *
+     * @return An instance of resolved DID DOC or null if DID is not found.
+     */
+    fun resolve(did: String): DIDDoc?
```

Comment:
Another option could be to return `Result<DIDDoc>` allowing the caller to manage any potential exceptions more gracefully (or creating their own exception hierarchy if so desired)

Reply:
In retrospect, if this has to be interoperable with java, maybe `Result` isn't such a good suggestion.  Maybe `Optional<DIDDoc>` might be more appropriate than just a `DIDDoc?` so the intention is clear in java-land